subsys/bluetooth/host/l2cap.c

```diff
@@ -88,6 +88,87 @@ struct bt_l2cap {
 static const struct bt_l2cap_ecred_cb *ecred_cb;
 static struct bt_l2cap bt_l2cap_pool[CONFIG_BT_MAX_CONN];
 
+void seg_destroy(struct net_buf *buf);
+#define SEGMENTS_COUNT CONFIG_BT_MAX_CONN
+/* see equivalent struct in conn.c */
+NET_BUF_POOL_FIXED_DEFINE(seg_pool, SEGMENTS_COUNT, 0,
+			  CONFIG_BT_CONN_TX_USER_DATA_SIZE, seg_destroy);
+
+struct seg_md {
+	struct bt_l2cap_le_chan *lechan;
+	struct bt_buf_view_meta view_meta;
+};
+
+struct seg_md seg_md_pool[SEGMENTS_COUNT];
+
+struct seg_md *get_seg_md(struct net_buf *seg)
+{
+	return &seg_md_pool[net_buf_id(seg)];
+}
+
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
+static void l2cap_chan_tx_resume(struct bt_l2cap_le_chan *ch);
+#endif
+
+void seg_destroy(struct net_buf *seg)
+{
+	/* Only relevant if there is segmentation going on. This is not possible
+	 * for LE ACL fixed channels, only for credit-based ones.
+	 */
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
```

Comment:
let's keep this for another PR. I can try to remove as many ifdefs as I can.
Because using `IS_ENABLED` is not that simple, I have to make dummy implementations for every called functions to satisfy the stupid build system.

Reply:
> Because using IS_ENABLED is not that simple, I have to make dummy implementations for every called functions to satisfy the stupid build system.

IS_ENABLED can only be used for functions that are compiled. If the functions are guarded by `#if defined` then we cannot use `IS_ENABLED` and this is the right approach. But yeah, ideally we'll reduce the number of `#if` we use overall -  Most of them are just guarding `static` data that the linker can remove anyways, so a lot of uses are unjust. Effectively we only need `#if` if we are accessing disabled Kconfig values or guarded struct fields